Sources/AngouriMath/Functions/Series.cs

```diff
@@ -44,5 +45,100 @@ internal static Entity TaylorExpansion(Entity expr, Variable exprVariable, Varia
             }
             throw new AngouriBugException($"We cannot get here, as {nameof(TaylorExpansionTerms)} is supposed to be endless");
         }
+
+        internal static IEnumerable<Entity> MultivariableTaylorExpansionTerms(Entity expr, (Variable exprVariable, Variable polyVariable, Entity point)[] exprToPolyVars)
+        {
+            // This structure is just a deconstruction of the terms that are added together in a multi taylor polynomial.
+            // Like n (x - a)^s (y - b)^t f'(x,y), where n is the coefficient, s and t are pointCoefficientDegrees, and then the partialDerivative
+            // See https://en.wikipedia.org/wiki/Taylor_series#Taylor_series_in_several_variables for the formula used here.
+            var lastTerms = new List<(int coefficient, int[] pointCoefficientDegrees, Entity partialDerivative)>();
+            var order = 0;
+            var variables = exprToPolyVars.Length;
+
+            while (true)
+            {
+                var newTerms = new List<(int coefficient, int[] pointCoefficientDegrees, Entity partialDerivative)>();
+
+                // Take a bunch of partial derivatives
+                if (order == 0) newTerms.Add((1, new int[variables], expr));
+                else {
+
+                    foreach (var lastTerm in lastTerms)
+                    {
+                        // with respect to each variable 
+                        for (int variableIndex = 0; variableIndex < exprToPolyVars.Length; variableIndex++)
+                        {
+                            var triple = exprToPolyVars[variableIndex];
+                            var newPointCoeffs = new int[variables];
+                            lastTerm.pointCoefficientDegrees.CopyTo(newPointCoeffs,0);
+                            newPointCoeffs[variableIndex] += 1;
+
+                            bool foundRepeatFlag = false;
+
+                            for (int newTermIndex = 0; newTermIndex < newTerms.Count; newTermIndex++)
+                            {
+                                var newTerm = newTerms[newTermIndex];
+
+                                if (Enumerable.SequenceEqual(newTerm.pointCoefficientDegrees, newPointCoeffs))
+                                {
+                                    newTerm.coefficient += lastTerm.coefficient;
+                                    foundRepeatFlag = true;
+                                    break;
+                                }
+                            }
+
+                            // If the partial derivative has already been computed, we just add to it's coefficient instead of duplicating it.
+                            if (!foundRepeatFlag)
+                            {
+                                var newPartialDerivative = lastTerm.partialDerivative.Differentiate(triple.exprVariable);
+                                newTerms.Add((lastTerm.coefficient, newPointCoeffs, newPartialDerivative));
+                            }
```

Comment:
Okay so is it to avoid computating `Fxy` when `Fyx` is already computed, right? Could you add a few comments explaning this chunk of code?

Also, I might be missing it, but do you multiply each partial derivative according to the coefficient, like for example when you need to multiply Fxy by two in `Fxx / 2 ... + 2Fxy / 2 ... + Fyy / 2...`? If yes, please somehow minimally explain it